ovn-org/ovn-kubernetes
# refactor: combined run, runCmd and runCmdWithEnvVars into an interface to enhance UT

Signed-off-by: Vishwanath Jayaraman <vjayaram@redhat.com>



**- What this PR does and why is it needed**

The methods run, runCmd and runWithEnvVars defined in go-controller/pkg/util/ovs.go that executed commands on OVN NB and SB databases is used widely by higher level functions (such as RunOVNAppctlWithTimeout, RunOVSVsctl, AddNormalActionOFFlow, ReplaceOFFlows, RunOVNSBAppCtl etc). To enable ease of writing Unit Test cases, it makes sense to group run, runCmd, runWithEnvVars into an interface definition and have mocks generated that would enable writing UT easy in other files that uses the above referenced higher level methods.

The mocks for third party(k8s.io/utils/exec) library methods such as `Run`, `SetEnv` etc,  used by run, runCmd and runCmdWithEnvVars are generated in the pkg/testing directory.

The mocks for the newly defined methods equivalent to `run`, `runCmd`, `runCmdWithEnvVars` that are grouped into an interface are generated in the `util` directory.

The tests in the pkg/util/ovs_unit_test.go where refactored to use the newly created mocks and coverage for some of the methods (DetectSCTPSupport and GetOVNDBServerInfo) have been enhanced

**- Special notes for reviewers**

As detailed in the above section, the changes are mainly in pkg/util/ovs.go and pkg/util/ovs_unit_test.go with third party library mocks restructured in pkg/testing directory.
The changes in the current PR are required to be reviewed and merged prior to writing UTs for additional files


**- How to verify it**

Pull this PR to your project workspace
navigate to ovn-kubernetes/go-controller/pkg/util directory
execute go test ./ovs.go ./ovs_unit_test.go -coverprofile cp.out 
execute go tool cover -html=cp.out (a browser will be launched with the coverage report displayed, check the coverage for the file ovs.go)


**- Description for the changelog**

combined run, runCmd and runCmdWithEnvVars functions in pkg/util/ovs.go into an interface to enhance UT
